sources/engine/Stride.Audio/Layers/OpenAL/AudioProvider.cs

```diff
@@ -255,7 +265,11 @@ public void SourceQueueBuffer(Source source, AudioBuffer buffer, nint pcm, int b
         buffer.Type = streamType;
         buffer.Size = bufferSize;
         al.BufferData(buffer.Value, source.Mono ? BufferFormat.Mono16 : BufferFormat.Stereo16, pcm.ToPointer(), bufferSize, source.SampleRate);
-        al.SourceQueueBuffers(source.Value, 1, &buffer.Value);
+        fixed (uint* bufferPtr = &buffer.Value)
+        {
+            al.SourceQueueBuffers(source.Value, 1, bufferPtr);
```

Comment:
The buffer will be used asynchronously, but it won't be pinned when it is used anymore, since the `fixed` statement is exited. To keep the buffer pinned, allocate it pinned, or from native memory, or use `GCHandle.Alloc(buffer.Value, GCHandleType.Pinned)` and then keep the `GCHandle` around and call `Free` on it once the buffer no longer needs to be pinned.

Reply:
Sorry for derailing the conversation slightly, but since we have you here, @ericwj can you quickly take a look at this comment we left a while ago on something you wrote https://github.com/stride3d/stride/pull/2393#discussion_r1682245735